pandas/core/index.py

```diff
@@ -1753,6 +1785,14 @@ def _get_nearest_indexer(self, target, limit):
         indexer = np.where(op(left_distances, right_distances)
                            | (right_indexer == -1),
                            left_indexer, right_indexer)
+        if tolerance is not None:
+            indexer = self._filter_indexer_tolerance(
+                target, indexer, tolerance)
+        return indexer
+
+    def _filter_indexer_tolerance(self, target, indexer, tolerance):
+        distance = abs(self.values[indexer] - target)
```

Comment:
Nope -- this is only called from inside `get_indexer`, which already calls `_convert_tolerance` at the top. The extra call to `_convert_tolerance` in `DatetimeIndex.get_loc` is to ensure that an appropriate error gets raised for a bad tolerance argument that cannot be coerced to a timedelta.